maps/src/MapBinner.cxx

```diff
@@ -260,14 +273,16 @@ MapBinner::Process(G3FramePtr frame, std::deque<G3FramePtr> &out)
 		stop_ = timestreams->GetStopTime();
 
 	if (!units_set_) {
-		T_->units = timestreams->GetUnits();
+		stub_->units = timestreams->GetUnits();
+		if (T_)
+			T_->units = timestreams->GetUnits();
 		if (Q_)
 			Q_->units = timestreams->GetUnits();
 		if (U_)
 			U_->units = timestreams->GetUnits();
 		units_set_ = true;
 	} else {
-		if (T_->units != timestreams->GetUnits())
+		if (stub_->units != timestreams->GetUnits())
```

Comment:
It seems fine to just not do this check if T_ isn't set. There also isn't any reason the weights-only case even needs to have timestreams (and it might actually be convenient if it did not require them for checking coverage on sim stubs)